### PR TITLE
[COOK-35] Add Tags

### PR DESCRIPTION
**Pull request by coronary**
## Description

Update adds tag functionality. Pulls existing tags from firestore and updates with new created tags

# [COOK-35](https://cdekalb.atlassian.net/browse/COOK-35)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] see tags on posts
- [x] add existing firestore tag to a post
- [x] add new tag and see change in firestore

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ import { ENV, DISCORD } from "./constants/constants";
 /* Store */
 import { Firebase, FirebaseContext } from "./firebase";
 import { Context } from "./store/Store";
-import { updateUser, updateStreams } from "./store/actions";
+import { updateUser, updateStreams, updateTags } from "./store/actions";
```
Comment (skbrown333) [inline src/App.tsx]: you can get rid of this i think

File: src/App.tsx [hunk 2]
```diff
@@ -47,6 +47,7 @@ export const App: FunctionComponent = () => {
         const user = await firebaseInstance.getCurrentUser();
         dispatch(updateUser(user));
         dispatch(updateStreams(await twitch.getStreams()));
+        // dispatch(updateTags(null));
```
Comment (skbrown333) [inline src/App.tsx]: remove

File: src/components/GuideDetailView/GuideDetailView.tsx [hunk 3]
```diff
@@ -27,9 +35,13 @@ import { Tag } from "../../models/Tag";
 /* Constants */
 import { newSection } from "../../constants/constants";
 
+/* Store */
```
Comment (skbrown333) [inline src/components/GuideDetailView/GuideDetailView.tsx]: remove

File: src/constants/constants.ts [hunk 4]
```diff
@@ -13,6 +13,10 @@ env.isLocal = isLocal;
 
 export const ENV = env;
 
+export const cookbook_id = "T4zKc3d28ITpz31iiRY3";
+import { Post } from "../models/Post";
```
Comment (skbrown333) [inline src/constants/constants.ts]: duplicate identifiers and ====

File: src/firebase/firebase.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import app from "firebase/app";
 import "firebase/auth";
 import "firebase/firestore";
+
```
Comment (skbrown333) [inline src/firebase/firebase.ts]: remove extra line

File: src/components/TagSection/TagSection.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  className: string;
+}
+
+export function TagSection(props) {
+  const firebase = useContext(FirebaseContext);
+  const [selectedOptions, setSelected] = useState(props.initial_tags);
+  const [options, setOptions] = useState(Array<any>());
+
+  const fetchTags = async () => {
+    let x = await firebase?.getTags();
```
Comment (skbrown333) [inline src/components/TagSection/TagSection.tsx:26]: instead of doing this do
```
let tags = x.map((doc) => {
     return doc.data().value
});
```

**Review by skbrown333 — CHANGES_REQUESTED** [review]
Looks really good! Just some small changes

File: src/store/initial-state.ts [hunk 7]
```diff
@@ -1,9 +1,11 @@
 export interface AppContext {
   user?: any;
   streams?: any;
+  tags?: any;
```
Comment (skbrown333) [inline src/store/initial-state.ts]: remove

File: src/store/initial-state.ts [hunk 8]
```diff
@@ -1,9 +1,11 @@
...
 }
 
 export const InitialState: AppContext = {
   user: null,
   streams: null,
+  tags: null,
```
Comment (skbrown333) [inline src/store/initial-state.ts]: remove

File: src/store/reducers.ts [hunk 9]
```diff
@@ -6,6 +6,8 @@ export const Reducer = (state, action) => {
       return { ...state, ...{ user: action.payload } };
     case Actions.UPDATE_STREAMS:
       return { ...state, streams: action.payload };
+    case Actions.UPDATE_TAGS:
```
Comment (skbrown333) [inline src/store/reducers.ts]: remove

File: src/components/TagSection/TagSection.tsx (hunk 6 above)
Comment (coronary) [reply src/components/TagSection/TagSection.tsx:26]: Since it's not actually an array it doesn't have the map function available

**Review by skbrown333 — DISMISSED** [review]

**Review by skbrown333 — APPROVED** [review]

**Pull request merged**